pkg/registry/core/service/storage/rest.go

```diff
@@ -261,8 +268,6 @@ func (rs *REST) Delete(ctx context.Context, id string, deleteValidation rest.Val
 		if err != nil && !errors.IsNotFound(err) {
 			return nil, false, err
 		}
-
-		rs.releaseAllocatedResources(svc)
```

Comment:
If I'm reading this correctly, the `endpoints.Delete` call should also be included in the AfterDelete handler, at which point I don't see a real reason for a custom REST#Delete implementation here.

The `rs.endpoints.Delete` and `rs.releaseAllocatedResources` calls are both done post-delete, which is already fragile if the server crashes between service delete and endpoints.Delete/releaseAllocatedResources calls, but I don't think moving these calls to AfterDelete makes that worse

Reply:
Actually, this Delete handler should still have been called at the point at which the Delete API call was made to the service. Why weren't the resources released at that point, even if the service object in etcd wasn't deleted until the finalizer was removed?